MyApp/app/policies/story_policy.rb

```diff
@@ -1,18 +1,19 @@
 class StoryPolicy < ApplicationPolicy
-class Scope
+  class Scope
     def initialize(user, scope)
       @user  = user
       @scope = scope
     end
 
     def resolve
-      if user.admin? || user.editor?
-        scope.all
-      else
-        scope.where(user: user)
+      if !(user.deleted_at == nil)
```

Comment:
it is good that you are thinking on every scenario, but I think It shouldn't be possible for user to make any action, it it is deleted